src/engineio/transport.rs

```diff
@@ -245,7 +246,7 @@ impl TransportClient {
             sender.send_dataframe(&WsDataFrame::new(
                 true,
                 websocket::dataframe::Opcode::Text,
-                encode_payload(vec![probe_packet]),
+                encode_payload(vec![probe_packet]).to_vec(),
```

Comment:
I'm not really happy with all those `to_vec` calls here as they copy the bytes, but looking at the `websocket` API I don't see any other way...